src/policy/markcompactspace.rs

```diff
@@ -0,0 +1,304 @@
+use super::space::{CommonSpace, Space, SpaceOptions, SFT};
+use crate::plan::MARKCOMPACT_CONSTRAINTS;
+use crate::util::constants::{BYTES_IN_WORD, MIN_OBJECT_SIZE};
+use crate::util::heap::layout::heap_layout::{Mmapper, VMMap};
+use crate::util::heap::{HeapMeta, MonotonePageResource, PageResource, VMRequest};
+use crate::util::metadata::load_metadata;
+use crate::util::metadata::side_metadata::{SideMetadataContext, SideMetadataSpec};
+use crate::util::metadata::{compare_exchange_metadata, extract_side_metadata};
+use crate::util::{alloc_bit, object_forwarding, Address, ObjectReference};
+use crate::{vm::*, TransitiveClosure};
+use atomic::Ordering;
+
+pub struct MarkCompactSpace<VM: VMBinding> {
+    common: CommonSpace<VM>,
+    pr: MonotonePageResource<VM>,
+    extra_header: usize,
```

Comment:
Also you would need to differentiate this field with `GC_EXTRA_HEADER_SIZE`. You can call it `header_reserve` or something. Otherwise it is confusing to see two variables with similar names and different values.